extensions/iceberg/src/main/java/io/deephaven/iceberg/util/IcebergTableWriter.java

```diff
@@ -260,6 +285,42 @@ private Map<String, Integer> readNameMappingDefault() {
         return nameMappingDefault;
     }
 
+    private List<SortColumn> computeSortColumns(@NotNull final SortOrder sortOrder, final boolean failOnUnmapped) {
+        if (sortOrder.isUnsorted()) {
+            return List.of();
+        }
+        final List<SortField> sortFields = sortOrder.fields();
+        final List<SortColumn> sortColumns = new ArrayList<>(sortFields.size());
+        for (final SortField sortField : sortOrder.fields()) {
+            final boolean ascending;
+            if (sortField.nullOrder() == NullOrder.NULLS_FIRST && sortField.direction() == SortDirection.ASC) {
+                ascending = true;
+            } else if (sortField.nullOrder() == NullOrder.NULLS_LAST && sortField.direction() == SortDirection.DESC) {
+                ascending = false;
+            } else {
+                if (failOnUnmapped) {
+                    throw new UnsupportedOperationException(
+                            "Cannot apply sort order " + sortOrder + " since Deephaven currently only supports " +
+                                    "sorting by {ASC, NULLS FIRST} or {DESC, NULLS LAST}");
+                }
+                return List.of();
+            }
+            final int fieldId = sortField.sourceId();
+            final String columnName = fieldIdToColumnName.get(fieldId);
+            if (columnName == null) {
+                if (failOnUnmapped) {
+                    throw new IllegalArgumentException("Cannot apply sort order " + sortOrder + " since column " +
+                            "corresponding to field ID " + fieldId + " not found in schema");
+                }
```

Comment:
We should figure out what the most appropriate exception type is; I think given this is at construction time, IAE is appropriate.